assets/shared/components/modal/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * WordPress dependencies
+ */
+import { Icon, close as closeIcon } from '@wordpress/icons';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Modal component.
+ *
+ * @param {Object}   props
+ * @param {Function} props.onClose  Callback to run when trying to close the modal.
+ * @param {string}   props.title    The title for the modal. Empty by default.
+ * @param {Object}   props.children The content of the modal.
+ * @return {JSX.Element} The modal component or null (if not open).
+ */
+const Modal = ( { onClose, title = '', children } ) => {
```

Comment:
Another thing that I realized while checking the Gutenberg Modal is that we're not using Portals here. I think it's important for modal implementations, in order to make it work in any place (if it's in a position relative, in a overflow hidden, or any other weird context).

See it here: https://github.com/WordPress/gutenberg/blob/v13.7.3/packages/components/src/modal/index.js#L124

Reply:
Added portals in here too: https://github.com/Automattic/sensei/commit/0c029ec980f35536f4fcad1481f15aa8773ba4f3